packages/financial-templates-lib/src/helpers/allowances.ts

```diff
@@ -0,0 +1,42 @@
+import { MAX_SAFE_ALLOWANCE, MAX_UINT_VAL, runTransaction } from "@uma/common";
+import { getAbi } from "@uma/contracts-node";
+import { ExpandedERC20Web3 } from "@uma/contracts-node";
+import type Web3 from "web3";
+import type { TransactionReceipt } from "web3-core";
+import type { GasEstimator } from "./GasEstimator";
+
+type ContractSendMethod = Parameters<typeof runTransaction>[0]["transaction"];
+
+// Sets `owner` allowance for `spender` to MAX_UINT_VAL, unless `spender` already has
+// an allowance > MAX_SAFE_ALLOWANCE. Return successful approval transaction data, or undefined
+// for skipped approvals.
+export const setAllowance = async (
+  web3: Web3,
+  gasEstimator: GasEstimator,
+  ownerAddress: string,
+  spenderAddress: string,
+  currencyAddress: string
+): Promise<
```

Comment:
I'm really not a huge fan of how some of this stuff is linted. This looks pretty janky. Not something we need to worry about right now in this PR but I'd be curious to explore other linter settings to improve this kind of layout for ts specific stuff.

Reply:
Agreed!